yaml/gateway_endpoint.go

```diff
@@ -0,0 +1,162 @@
+package yaml
+
+import (
+	"fmt"
+
+	"github.com/buildwithgrove/path/envoy/auth_server/proto"
+
+	grpc_server "github.com/buildwithgrove/path-auth-data-server/grpc"
+)
+
+/* ----------------------------- GatewayEndpoint YAML Struct ----------------------------- */
+
+type (
+	// gatewayEndpointYAML represents the structure of a single GatewayEndpoint in the YAML file.
+	gatewayEndpointYAML struct {
+		Auth         authYAML          `yaml:"auth"`
+		RateLimiting rateLimitingYAML  `yaml:"rate_limiting"`
+		Metadata     map[string]string `yaml:"metadata"`
+	}
+	// authYAML represents the Auth section of a single GatewayEndpoint in the YAML file.
+	authYAML struct {
+		AuthType           grpc_server.AuthType `yaml:"auth_type"`
+		APIKey             *string              `yaml:"api_key,omitempty"`
+		JWTAuthorizedUsers []string             `yaml:"jwt_authorized_users,omitempty"`
+	}
+	// rateLimitingYAML represents the RateLimiting section of a single GatewayEndpoint in the YAML file.
+	rateLimitingYAML struct {
+		ThroughputLimit     int                             `yaml:"throughput_limit"`
```

Comment:
Do you mind copy-pasting your comments from elsewhere w/ a few details on what this is?